File: spring-api/src/main/java/com/tenutz/cracknotifier/web/api/dto/crack/CracksResponse.java
package com.tenutz.cracknotifier.web.api.dto.crack;

import com.tenutz.cracknotifier.domain.crack.Crack;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class CracksResponse {

    private String seq;
    private String imageUrl;
    private String createdAt;
    private float accuracy;
    private String region3DepthName;

    public CracksResponse(Crack crack) {
        this.seq = crack.getSeq();
        this.imageUrl = crack.getImage();
        this.createdAt = crack.getCreatedAt().toString();
        this.accuracy = crack.getAccuracy();
        this.region3DepthName = crack.getAddress().getRegion3DepthName();
    }
}

File: android/app/src/main/java/com/tenutz/cracknotifier/data/api/dto/crack/CracksResponse.kt
package com.tenutz.cracknotifier.data.api.dto.crack

import java.util.*

data class CracksResponse(
    val seq: String,
    val imageUrl: String? = null,
    val createdAt: Date? = null,
    val accuracy: Float,
    val region3DepthName: String? = null,
)